scripts/apps/authoring-react/authoring-angular-integration.tsx

```diff
@@ -184,6 +206,97 @@ function getInlineToolbarActions(
         return actions;
     };
 
+    const correctAction = (text: string = gettext('SEND CORRECTION'), tooltip?: string): ITopBarWidget<IArticle> => ({
+        group: 'end',
+        priority: 0.1,
+        component: () => (
+            <Button
+                tooltip={tooltip}
+                text={text}
+                style="filled"
+                type="primary"
+                onClick={() => {
+                    handleUnsavedChanges()
+                        .then(() => sdApi.article.publishItem(item, getLatestItem(), 'correct'))
+                        .then(() => initiateClosing());
+                }}
+            />
+        ),
+        availableOffline: false,
+    });
+
+    const killAction: ITopBarWidget<IArticle> = {
+        group: 'end',
+        priority: 0.1,
+        component: () => (
+            <Button
+                text={gettext('SEND KILL')}
+                style="filled"
+                type="primary"
+                onClick={() => {
+                    handleUnsavedChanges()
+                        .then(() => sdApi.article.publishItem(item, getLatestItem(), 'kill'))
+                        .then(() => initiateClosing());
+                }}
+            />
+        ),
+        availableOffline: false,
+    };
+
+    const takedownAction: ITopBarWidget<IArticle> = {
+        group: 'end',
+        priority: 0.1,
+        component: () => (
+            <Button
+                text="T"
+                tooltip={gettext('Takedown')}
+                style="filled"
+                type="primary"
+                onClick={() => {
+                    handleUnsavedChanges()
+                        .then(() => sdApi.article.publishItem(item, getLatestItem(), 'takedown'))
+                        .then(() => initiateClosing());
+                }}
+            />
+        ),
+        availableOffline: false,
+    };
+
+    const unpublishAction: ITopBarWidget<IArticle> = {
+        group: 'end',
+        priority: 0.1,
+        component: () => (
+            <Button
+                tooltip={gettext('UNPUBLISH')}
+                text={'UP'}
+                style="filled"
+                type="primary"
+                onClick={() => {
+                    handleUnsavedChanges()
+                        .then(() => sdApi.article.publishItem(item, getLatestItem(), 'unpublish'))
+                        .then(() => initiateClosing());
+                }}
+            />
+        ),
+        availableOffline: false,
+    };
+
+    const closeAuthoringAction: ITopBarWidget<IArticle> = {
+        group: 'end',
+        priority: 0.1,
+        component: () => (
+            <Button
+                text={gettext('CANCEL CORRECTION')}
+                style="filled"
+                type="primary"
+                onClick={() => {
+                    initiateClosing();
+                }}
+            />
+        ),
+        availableOffline: false,
+    };
+
     switch (itemState) {
```

Comment:
did you go through existing authoring to double check which actions have to be available in each situation?